code-for-canada/project-thundercat
# [Login Frontend] Introducing the registration flow

# Description

This PR is introducing the registration flow, meaning that we are now able to create new users from the registration form and login them.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Registration Form:**

![image](https://user-images.githubusercontent.com/23021242/57800951-a3aa0c00-7720-11e9-85d3-9576cd192f18.png)

**Creating a new account:**

![image](https://user-images.githubusercontent.com/23021242/57801001-bd4b5380-7720-11e9-97eb-d5fb24c50215.png)

**Account Created:**

![image](https://user-images.githubusercontent.com/23021242/57801046-d81dc800-7720-11e9-9a0c-e9a77ad5fef8.png)

**If the user (email address) already exists in the system:**

![image](https://user-images.githubusercontent.com/23021242/57801092-f71c5a00-7720-11e9-9948-95a778904186.png)

**Login in with the new user credentials:**

![image](https://user-images.githubusercontent.com/23021242/57801141-11563800-7721-11e9-8793-9996c30b712e.png)

![image](https://user-images.githubusercontent.com/23021242/57801158-174c1900-7721-11e9-937a-d2211169ac03.png)

# Testing

Manual steps to reproduce this functionality:

1.  Create a new account by using the registration form.
2.  Make sure that you are able to login once the new account has been created.
3.  Try to create the exact same account and make sure that you receive the _already existing account_ error.

# Checklist

Applicable for all code changes.
